### PR TITLE
AV-187265 : SRP provenance 3.0 changes

### PR DESCRIPTION
**Pull request by jeyapradeen-avi**

### CODE REVIEW

**Comment by akshayhavile** [comment]
Can one of the admins verify this patch?

File: hack/jenkins/save_build.sh [hunk 1]
```diff
@@ -57,24 +58,77 @@ JENKINS_INSTANCE=$(echo $JENKINS_URL | sed -E 's/^\s*.*:\/\///g' | sed -E 's/:.*
...
+# initialize blank provenance in the working directory, $SRP_WORKING_DIR
+sudo /srp-tools/srp provenance init --working-dir $WORKSPACE/provenance
+sudo /srp-tools/srp provenance add-build jenkins --instance $JENKINS_INSTANCE --build-number $BUILD_NUMBER --job-name $JENKINS_JOB_NAME --working-dir $WORKSPACE/provenance
+
+# add an action for the golang build, importing the observations that were captured in the build-golang-app step
+sudo /srp-tools/srp provenance action start --name=amko-build --working-dir $WORKSPACE/provenance
+sudo /srp-tools/srp provenance action import-observation --name=amko-obs --file=$WORKSPACE/provenance/network_provenance.json --working-dir $WORKSPACE/provenance
+sudo /srp-tools/srp provenance action stop --working-dir $WORKSPACE/provenance
+
+# Add an action for make build itself.  Note that it does not actually record the real command line here.  Also note
```
Comment (akshayhavile) [inline hack/jenkins/save_build.sh:73]: please delete commented code.

Comment (jeyapradeen-avi) [reply hack/jenkins/save_build.sh:73]: @akshayhavile , these are description which tell what we have to do in the next steps. we need this in the file

File: hack/jenkins/save_build.sh [hunk 2]
```diff
@@ -57,24 +58,77 @@ JENKINS_INSTANCE=$(echo $JENKINS_URL | sed -E 's/^\s*.*:\/\///g' | sed -E 's/:.*
...
-    if [ -f  $provenance_source_file ]
-    then
-        source_provenance --append
-    else
-        source_provenance
-    fi
-done
-cd $WORKSPACE
-sudo /srp-tools/srp provenance merge --source ./provenance/source.json --network ./provenance/provenance.json --saveto ./provenance/merged.json
+# Function to run srp source provenance command
```
Comment (akshayhavile) [inline hack/jenkins/save_build.sh:114]: @jeyapradeen-avi: Code from line number 114 to 131 is required?

**Comment by jeyapradeen-avi** [comment]
This changed taken care by [Jenkins file ](https://github.com/vmware/global-load-balancing-services-for-kubernetes/commit/cf364668ca3a76290fbbd82c7b8eb60938e5b574)

**Pull request closed without merging**